packages/auth/src/providers/cognito/utils/clients/ConfirmSignUpClient.ts

```
@@ -0,0 +1,30 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import type {
+	ConfirmSignUpCommandInput,
+	ConfirmSignUpCommandOutput,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { UserPoolHttpClient } from './HttpClients';
+import { UserPoolClient } from './UserPoolClient';
+
+export type ConfirmSignUpClientInput = Pick<
+	ConfirmSignUpCommandInput,
+	| 'Username'
+	| 'ClientMetadata'
+	| 'ConfirmationCode'
+	| 'ForceAliasCreation'
+	| 'UserContextData'
+>;
+
+export async function confirmSignUpClient(
+	params: ConfirmSignUpClientInput
+): Promise<ConfirmSignUpCommandOutput> {
+	const client = new UserPoolHttpClient(UserPoolClient.region);
+	const result: ConfirmSignUpCommandOutput =
+		await client.send<ConfirmSignUpCommandOutput>('ConfirmSignUp', {
```

I know we haven't been doing it in the other APIs as well but should we be swallowing and retry for some of the exceptions and only rethrow ones that makes sense for customers to know about? 

https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ConfirmSignUp.html#CognitoUserPools-ConfirmSignUp-request-AnalyticsMetadata:~:text=empty%20HTTP%20body.-,Errors,-For%20information%20about

---

The good thing is that `confirmSignUp` is retriable if a there is a code mismatch. So not something that customers will be blocked on confirming their accounts.